src/EditorFeatures/CSharpTest/Completion/CompletionProviders/XmlDocumentationCommentCompletionProviderTests.cs

```diff
@@ -1084,6 +1134,93 @@ static void Main(string[] args)
 ", "args");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task ParamNames()
+        {
+            await VerifyItemExistsAsync(@"
+/// <param name=""$$""/>
+static void Foo(string str)
+{
+}
+", "str");
+        }
+
+        [WorkItem(17872, "https://github.com/dotnet/roslyn/issues/17872")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TypeParamRefNames()
+        {
+            var text = @"
+public class Outer<TOuter>
+{
+    public class Inner<TInner>
+    {
+        /// <summary>
+        /// <typeparamref name=""$$""/>
+        /// </summary>
+        public int Method<TMethod>(T green) { }
+    }
+}";
+
+            await VerifyItemsExistAsync(text, "TOuter", "TInner", "TMethod");
+        }
+
+        [WorkItem(17872, "https://github.com/dotnet/roslyn/issues/17872")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TypeParamRefNamesPartiallyTyped()
+        {
+            var text = @"
+public class Outer<TOuter>
+{
+    public class Inner<TInner>
+    {
+        /// <summary>
+        /// <typeparamref name=""T$$""/>
+        /// </summary>
+        public int Method<TMethod>(T green) { }
+    }
+}";
+
+            await VerifyItemsExistAsync(text, "TOuter", "TInner", "TMethod");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TypeParamNames()
+        {
+            var text = @"
+public class Outer<TOuter>
+{
+    public class Inner<TInner>
+    {
+        /// <summary>
+        /// <typeparam name=""$$""/>
+        /// </summary>
+        public int Method<TMethod>(T green) { }
+    }
+}";
+
+            await VerifyItemsExistAsync(text, "TMethod");
```

Comment:
😕 This is strange. `<typeparam>` is a top-level element but here you're testing it inside of a `<summary>` element. I suppose it's fine to offer completion in the scenario but hopefully we would provide a warning somewhere else regarding the use of the element here.